providers/implementations/kdfs/hkdf.c

```diff
@@ -496,3 +527,216 @@ static int HKDF_Expand(const EVP_MD *evp_md,
     HMAC_CTX_free(hmac);
     return ret;
 }
+
+/*
+ * TLS uses slight variations of the above and for FIPS validation purposes,
+ * they need to be present here.
+ * Refer to RFC 8446 section 7 for specific details.
+ */
+
+/*
+ * Given a |secret|; a |label| of length |labellen|; and |data| of length
+ * |datalen| (e.g. typically a hash of the handshake messages), derive a new
+ * secret |outlen| bytes long and store it in the location pointed to be |out|.
+ * The |data| value may be zero length. Any errors will be treated as fatal if
+ * |fatal| is set. Returns 1 on success  0 on failure.
+ */
+static int prov_tls13_hkdf_expand(const EVP_MD *md,
+                                  const unsigned char *key, size_t keylen,
+                                  const unsigned char *prefix, size_t prefixlen,
+                                  const unsigned char *label, size_t labellen,
+                                  const unsigned char *data, size_t datalen,
+                                  unsigned char *out, size_t outlen)
+{
+    size_t hkdflabellen;
+    /*
+     * 2 bytes for length of derived secret + 1 byte for length of combined
+     * prefix and label + bytes for the label itself + 1 byte length of hash
```

Comment:
does "label itself" include the prefix?
Maybe we don't need the comment at this level of detail...

Reply:
It's a really good comment, but you are correct it's not critical anymore.

Reply:
It might actually go better above the WPACKET stanza, come to think of it...

Reply:
Okay, I've put it there and slightly reworded the final sentence.